pandas/tests/reshape/test_merge.py

```diff
@@ -724,6 +724,63 @@ def test_indicator(self):
                           how='outer', indicator=True)
         assert_frame_equal(test5, hand_coded_result)
 
+    def test_validation(self):
```

Comment:
The only issue I see is that there are requirements for `merge_as` that don't apply to `merge` (for example, `merge_as` requires sorting). I can do this, but will limit set of tests I can run to those that work on both (current tests for `merge` don't work for `merge_asof`)